includes/Gateway/Cash_App_Pay_Gateway.php

```diff
@@ -0,0 +1,1226 @@
+				'pending' => false,
+			);
+
+			$line_items[] = $item;
+		}
+
+		if ( $totals['shipping'] > 0 ) {
+			$line_items[] = array(
+				'label'   => __( 'Shipping', 'woocommerce-square' ),
+				'amount'  => number_format( $totals['shipping'], 2, '.', '' ),
+				'pending' => false,
+			);
+		}
+
+		if ( $totals['taxes'] > 0 ) {
+			$line_items[] = array(
+				'label'   => __( 'Tax', 'woocommerce-square' ),
+				'amount'  => number_format( $totals['taxes'], 2, '.', '' ),
+				'pending' => false,
+			);
+		}
+
+		if ( $totals['discount'] > 0 ) {
+			$line_items[] = array(
+				'label'   => __( 'Discount', 'woocommerce-square' ),
+				'amount'  => number_format( $totals['discount'], 2, '.', '' ),
+				'pending' => false,
+			);
+		}
+
+		if ( $totals['fees'] > 0 ) {
+			$line_items[] = array(
+				'label'   => __( 'Fees', 'woocommerce-square' ),
+				'amount'  => number_format( $totals['fees'], 2, '.', '' ),
+				'pending' => false,
+			);
+		}
+
+		return $line_items;
+	}
+
+	/**
+	 * Get the payment request object in an ajax request
+	 *
+	 * @since x.x.x
+	 * @return void
+	 */
+	public function ajax_get_payment_request() {
+		check_ajax_referer( 'wc-cash-app-get-payment-request', 'security' );
+
+		$payment_request = array();
+
+		try {
+			$payment_request = $this->get_payment_request();
+
+			if ( empty( $payment_request ) ) {
+				/* translators: Context (product, cart, checkout or page) */
+				throw new \Exception( esc_html__( 'Empty payment request data for page.', 'woocommerce-square' ) );
+			}
+		} catch ( \Exception $e ) {
+			wp_send_json_error( $e->getMessage() );
+		}
+
+		wp_send_json_success( wp_json_encode( $payment_request ) );
+	}
+
+	/**
+	 * Set continuation session to select the cash app payment method after the redirect back from the cash app
+	 *
+	 * @since x.x.x
+	 * @return void
+	 */
+	public function ajax_set_continuation_session() {
+		check_ajax_referer( 'wc-cash-app-set-continuation-session', 'security' );
+		$clear_session = ( isset( $_POST['clear'] ) && 'true' === sanitize_text_field( wp_unslash( $_POST['clear'] ) ) );
+
+		try {
+			if ( $clear_session ) {
+				WC()->session->set( 'wc_square_cash_app_pay_continuation', null );
+			} else {
+				WC()->session->set( 'wc_square_cash_app_pay_continuation', 'yes' );
+			}
+		} catch ( \Exception $e ) {
+			wp_send_json_error( $e->getMessage() );
+		}
+
+		wp_send_json_success( wp_json_encode( array( 'success' => true ) ) );
+	}
+
+	/**
+	 * Determines if the current request is a continuation of a cash app pay payment.
+	 *
+	 * @return boolean
+	 */
+	public function is_cash_app_pay_continuation() {
+		return WC()->session && 'yes' === WC()->session->get( 'wc_square_cash_app_pay_continuation' );
+	}
+
+	/**
+	 * Returns cart totals in an array format
+	 *
+	 * @since x.x.x
+	 * @throws \Exception if no cart is found
+	 * @return array
+	 */
+	public function get_cart_totals() {
+		if ( ! isset( WC()->cart ) ) {
+			throw new \Exception( 'Cart data cannot be found.' );
```

Comment:
Is this added to a log file or shown directly to a shopper?

Reply:
This message will be displayed in the browser console (log) if the debug mode on the checkout page is enabled, and it will not be shown directly to the shopper. No error will be logged if the debug mode is OFF.